src/loader.rs

```diff
@@ -90,18 +122,53 @@ impl MarkedEventReceiver for YamlLoader {
                     // Datatype is not specified, or unrecognized
                     Yaml::from_str(&v)
                 };
-
-                self.insert_new_node((node, aid));
+                self.insert_new_node((Node::from_bare_yaml(node).with_marker(marker), aid));
             }
             Event::Alias(id) => {
                 let n = match self.anchor_map.get(&id) {
                     Some(v) => v.clone(),
-                    None => Yaml::BadValue,
+                    None => Node::from_bare_yaml(Yaml::BadValue),
                 };
-                self.insert_new_node((n, 0));
+                self.insert_new_node((n.with_marker(marker), 0));
+            }
+        }
+    }
+}
+
+impl<Node> YamlLoader<Node>
+where
+    Node: LoadableYamlNode,
+{
+    fn insert_new_node(&mut self, node: (Node, usize)) {
+        // valid anchor id starts from 1
+        if node.1 > 0 {
+            self.anchor_map.insert(node.1, node.0.clone());
+        }
+        if self.doc_stack.is_empty() {
+            self.doc_stack.push(node);
+        } else {
+            let parent = self.doc_stack.last_mut().unwrap();
```

Comment:
Nitpick: you can avoid the unwrap by reorganizing the if-then-else:

```rust
if let Some(parent) = self.doc_stack.last_mut() {
  //...code of the current else...
}
else {
    self.doc_stack.push(node);
}
```

Reply:
I like nitpicks like these. I rely too much on `unwrap` sometimes. Good catch!